dashboard/test/ui/features/platform/signing_in.feature

```diff
@@ -69,3 +70,18 @@ Scenario: Join existing section from sign in page goes to section join page
   And I click ".section-sign-in button" to load a new page
   Then I wait to see ".main"
   And element ".main" contains text "Register to join section"
+
+Scenario: Student navigates to provided cached level link with a login_required parameter
+  Given I create a student named "Carah Student"
+  And I sign out
+  Given I am on "http://studio.code.org/s/poem-art-2021/lessons/1/levels/1?login_required=true"
+  Then I wait until I am on "http://studio.code.org/users/sign_in"
+  And I wait to see "#signin"
+  And I fill in username and password for "Carah Student"
+  And I click "#signin-button" to load a new page
+  Then I wait until I am on "http://studio.code.org/s/poem-art-2021/lessons/1/levels/1"
+
+Scenario: Student already logged in navigates to provided cached level link with a login_required parameter
+  Given I create a student who has never signed in named "François Student" and go home
+  And I am on "http://studio.code.org/s/poem-art-2021/lessons/1/levels/1?login_required=true"
+  Then I wait until I am on "http://studio.code.org/s/poem-art-2021/lessons/1/levels/1"
```

Comment:
These tests both fail with the original code before this PR patch.

Reply:
![image](https://github.com/code-dot-org/code-dot-org/assets/31674/a6d03f60-45bd-407c-8b8a-dd0928f91186)

Thankfully, I added some context around a timeout to show me what page it is actually on... and it doesn't redirect.

Reply:
Same for when the student is signed in:

![image](https://github.com/code-dot-org/code-dot-org/assets/31674/553c8c9c-f121-4d84-a094-df10faac7665)

Reply:
This is because that level isn't seeded on drone. I've changed it to the star wars level which is a cached level and seeded in the test environment